ktor-features/ktor-metrics-micrometer-prometheus/jvm/src/io/ktor/metrics/micrometer/prometheus/installFeatureWithAuth.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2014-2019 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.metrics.micrometer.prometheus
+
+import io.ktor.application.*
+import io.ktor.auth.*
+import io.ktor.routing.*
+
+fun Application.installMetricsEndpointWithAuth(config: MicrometerMetricsPrometheus.Configuration) {
+
+    checkIfAuthenticationFeatureIsInstalled()
+
+    routing {
+        authenticate(*config.authenticationConfig) {
+            metricsEndpoint(config)
+        }
+    }
+}
+
+private fun Application.checkIfAuthenticationFeatureIsInstalled() {
+    try {
+        feature(Authentication)
+    } catch (ex: MissingApplicationFeatureException) {
+        throw Exception(
```

Comment:
Why `Exception`? I'd prefer to see some meaningful type

Reply:
fixed (I forgot how easy it is to declare a class in Kotlin...) In general this exception should not be caught during runtime, it shows the developer that they did not configure the feature correctly.